core/capabilities/gateway_connector/service_wrapper.go

```diff
@@ -0,0 +1,156 @@
+
+func NewConnectorSigner(config config.GatewayConnector, signerKey *ecdsa.PrivateKey, lggr logger.Logger) (*connectorSigner, error) {
+	return &connectorSigner{
+		nodeAddress: config.NodeAddress(),
+		signerKey:   signerKey,
+		lggr:        lggr.Named("ConnectorSigner"),
+	}, nil
+}
+
+func (h *connectorSigner) Sign(data ...[]byte) ([]byte, error) {
+	return gwCommon.SignData(h.signerKey, data...)
+}
+
+func (h *connectorSigner) HandleGatewayMessage(ctx context.Context, gatewayID string, msg *api.Message) {
+}
+
+func (h *connectorSigner) Start(ctx context.Context) error {
+	return h.StartOnce("ConnectorSigner", func() error {
+		return nil
+	})
+}
+func (h *connectorSigner) Close() error {
+	return h.StopOnce("ConnectorSigner", func() (err error) {
+		return nil
+	})
+}
+
+func (h *connectorSigner) SetConnector(connector connector.GatewayConnector) {
+	h.connector = connector
+}
+
+func translateConfigs(f config.GatewayConnector) connector.ConnectorConfig {
+	r := connector.ConnectorConfig{}
+	r.NodeAddress = f.NodeAddress()
+	r.DonId = f.DonID()
+
+	if len(f.Gateways()) != 0 {
+		r.Gateways = make([]connector.ConnectorGatewayConfig, len(f.Gateways()))
+		for index, element := range f.Gateways() {
+			r.Gateways[index] = connector.ConnectorGatewayConfig{Id: element.ID(), URL: element.URL()}
+		}
+	}
+
+	r.WsClientConfig = network.WebSocketClientConfig{HandshakeTimeoutMillis: f.WSHandshakeTimeoutMillis()}
+	r.AuthMinChallengeLen = f.AuthMinChallengeLen()
+	r.AuthTimestampToleranceSec = f.AuthTimestampToleranceSec()
+	return r
+}
+
+// NOTE: this wrapper is needed to make sure that our services are started after Keystore.
+func NewGatewayConnectorServiceWrapper(config config.GatewayConnector, keystore keystore.Eth, lggr logger.Logger) *ServiceWrapper {
+	return &ServiceWrapper{
+		config:   config,
+		keystore: keystore,
+		lggr:     lggr,
+	}
+}
+
+func (e *ServiceWrapper) Start(ctx context.Context) error {
+	return e.StartOnce("GatewayConnectorServiceWrapper", func() error {
+		conf := e.config
+		chainID, _ := new(big.Int).SetString(conf.ChainIDForNodeKey(), 0)
+		enabledKeys, err := e.keystore.EnabledKeysForChain(ctx, chainID)
+		if err != nil {
+			return err
+		}
+		if len(enabledKeys) == 0 {
+			return errors.New("no available keys found")
+		}
+		configuredNodeAddress := common.HexToAddress(conf.NodeAddress())
+		idx := slices.IndexFunc(enabledKeys, func(key ethkey.KeyV2) bool { return key.Address == configuredNodeAddress })
+
+		if idx == -1 {
+			return errors.New("key for configured node address not found")
+		}
+		signerKey := enabledKeys[idx].ToEcdsaPrivKey()
+		if enabledKeys[idx].ID() != conf.NodeAddress() {
+			return errors.New("node address mismatch")
+		}
+
+		signer, err := NewConnectorSigner(e.config, signerKey, e.lggr)
+		if err != nil {
+			return err
+		}
+		translated := translateConfigs(conf)
+		e.connector, err = connector.NewGatewayConnector(&translated, signer, clockwork.NewRealClock(), e.lggr)
```

Comment:
Instead of using a real clock here, we may want to use a parameterized clock, such as keeping one on `ServiceWrapper`. This way we can pass a fake clock for testing.

Reply:
@bolekk your thoughts?

Reply:
makes sense to me